### ISSUE TITLE
Carthage support

### ISSUE THREAD

**Issue by z4r** [kasketis/netfox#26]
Hi,
Very good work.

Could you please share your schema to support carthage installation?

https://github.com/Carthage/Carthage/blob/master/README.md#supporting-carthage-for-your-framework

THX

**Comment by kasketis** [comment]
Hello @z4r , 
Thanks for your kind words :)
I have uploaded the .xcodeproj which includes a shared scheme on it. Could you please check if you can install netfox via carthage now?

**Comment by z4r** [comment]
Hi @kasketis ,
You need to mark as "shared" that schema.

From the menu Product > Scheme > Manage Schemes

And from this panel mark the schema as "shared" (the last column).

**Comment by kasketis** [comment]
Hi again @z4r ,
The schema is already marked as "shared" as you can see in the below image
<img width="1021" alt="screen shot 2015-11-29 at 11 50 37 am" src="https://cloud.githubusercontent.com/assets/2895107/11456376/7771dce6-968f-11e5-848e-90eb2b64d34c.png">
Is there any error message on carthage?

**Comment by kasketis** [comment]
I succefully managed to create the netfox.framework through Carthage
<img width="648" alt="screen shot 2015-11-29 at 12 42 53 pm" src="https://cloud.githubusercontent.com/assets/2895107/11456603/ebd1c414-9696-11e5-94c6-1f69f9447321.png">
Please give it a try again

**Comment by michaelfransen** [comment]
I just tried and can't install via carthage. 

```
*** Building scheme "netfox" in netfox.xcodeproj
2015-11-30 11:22:12.917 xcodebuild[20466:799929] [MT] PluginLoading: Required plug-in compatibility UUID 7265231C-39B4-402C-89E1-16167C4CC990 for plug-in at path '~/Library/Application Support/Developer/Shared/Xcode/Plug-ins/OMQuickHelp.xcplugin' not present in DVTPlugInCompatibilityUUIDs
2015-11-30 11:22:12.919 xcodebuild[20466:799929] [MT] PluginLoading: Required plug-in compatibility UUID 7265231C-39B4-402C-89E1-16167C4CC990 for plug-in at path '~/Library/Application Support/Developer/Shared/Xcode/Plug-ins/Alcatraz.xcplugin' not present in DVTPlugInCompatibilityUUIDs
** BUILD FAILED **
```

**Comment by kasketis** [comment]
@michaelfransen did you manage to install it successfully after your changes in #30 ?

**Comment by michaelfransen** [comment]
This did not fix that. I have some time right now so I am going to look into this.

**Comment by michaelfransen** [comment]
Just confirmed that Carthage works after my pull request. However, there are still warnings. I will keep looking into it.

**Comment by kasketis** [comment]
Thanks @michaelfransen ! I appreciate your help :+1: 

After NFXStatisticsController's add to the project, I get this
<pre>
kasketiss-MBP:netfox kasketis$ carthage build --no-skip-current
*** xcodebuild output can be found in /var/folders/5n/zvmyn6753_37_qryxlf42ych0000gn/T/carthage-xcodebuild.WRRs2Q.log
*** Building scheme "netfox" in netfox.xcodeproj
2015-11-30 21:15:39.254 xcodebuild[40822:1167194] [MT] PluginLoading: Required plug-in compatibility UUID 7265231C-39B4-402C-89E1-16167C4CC990 for plug-in at path '~/Library/Application Support/Developer/Shared/Xcode/Plug-ins/Unity4XC.xcplugin' not present in DVTPlugInCompatibilityUUIDs
2015-11-30 21:15:41.370 xcodebuild[40844:1167300] [MT] PluginLoading: Required plug-in compatibility UUID 7265231C-39B4-402C-89E1-16167C4CC990 for plug-in at path '~/Library/Application Support/Developer/Shared/Xcode/Plug-ins/Unity4XC.xcplugin' not present in DVTPlugInCompatibilityUUIDs
</pre>

The xcodebuild log shows that everything went as expected
<pre>
kasketiss-MBP:netfox kasketis$ cat /var/folders/5n/zvmyn6753_37_qryxlf42ych0000gn/T/carthage-xcodebuild.WRRs2Q.log
Build settings from command line:
    ONLY_ACTIVE_ARCH = NO
    SDKROOT = iphonesimulator9.1

=== BUILD TARGET netfox OF PROJECT netfox WITH CONFIGURATION Release ===

Check dependencies

Write auxiliary files
write-file /Users/kasketis/Library/Developer/Xcode/DerivedData/netfox-gbpnwpcvooalsffjtbmnusilbxjo/Build/Intermediates/netfox.build/all-product-headers.yaml

** BUILD SUCCEEDED **

Build settings from command line:
    BITCODE_GENERATION_MODE = bitcode
    ONLY_ACTIVE_ARCH = NO
    SDKROOT = iphoneos9.1

=== BUILD TARGET netfox OF PROJECT netfox WITH CONFIGURATION Release ===

Check dependencies

Write auxiliary files
write-file /Users/kasketis/Library/Developer/Xcode/DerivedData/netfox-gbpnwpcvooalsffjtbmnusilbxjo/Build/Intermediates/netfox.build/all-product-headers.yaml

** BUILD SUCCEEDED **
</pre>

but I'm getting some errors when I'm trying to import the produced netfox.framework in a demo project.
Does the framework work correctly in your project?

**Comment by michaelfransen** [comment]
The framework produced from Carthage isn't working for me either. All the header files except for one are missing.

**Comment by kasketis** [comment]
I managed to get it work! Please follow the next steps:
- Drag & drop the framework in your project (Check "Copy items if needed" and "Create groups")
- Add the framework in "Embedded binaries" section of your target's General tab
- Add "import netfox" in your AppDelegate file
- Then use NFX.sharedInstance().start()

My produced framework looks like that
<img width="602" alt="screen shot 2015-11-30 at 9 57 43 pm" src="https://cloud.githubusercontent.com/assets/2895107/11482783/7b735a76-97ad-11e5-9fc0-b1aa01802ca8.png">

I'm not a Carthage user but I think that's not the "Carthage" way to import libraries. 
What's the "official way" for that to be done? Must I create a "Cartfile"?

**Comment by michaelfransen** [comment]
I just got it to work with Carthage. YAY!. I added the framework to my project and then made a copy build phase to copy the framework. It works now. I would like to figure out why it generates those symbol maps though, because none of my other frameworks do. 

Also, I noticed in the build settings of netfox, that the "contains swift code" is set to no. That should be set to yes, right?

**Comment by kasketis** [comment]
Nice :+1:  :beer: Could you please post your steps?

As for "Contains swift code" flag I think it's not mandatory to be YES in the framework (but of course I can set it). Have you included other swift projects via Carthage? If yes could you please check their project settings?

**Comment by michaelfransen** [comment]
You are correct. "Contains swift code" is not required.

**Comment by michaelfransen** [comment]
Here were the steps I used to get Carthage working.

1. Add ```github "kasketis/netfox"``` to your Cartfile. (Note, https://github.com/kasketis/netfox/pull/307 needs to be include first). 
1. Run ```carthage update```
1. Drag ```netfox.framework``` from ```Carthage/Build/[platform]/``` into you Linked Fameworks and Libraries under your target. I chose to not copy. 
1. I also created a new copy script build phase as the last step. I choose frameworks as the destination. I then clicked the plus sign and selected the netfox framework.
